src/mds/MDSRank.cc

```diff
@@ -1758,6 +1758,13 @@ bool MDSRankDispatcher::handle_asok_command(
     } else {
       mdcache->dump_cache(path);
     }
+  } else if (command == "dump below") {
+    string root;
+    int64_t depth;
+    cmd_getval(g_ceph_context, cmdmap, "root", root);
+    if (!cmd_getval(g_ceph_context, cmdmap, "depth", depth))
+      depth = -1;
+    mdcache->dump_cache(root, depth, f);
```

Comment:
If metadata ops are in flight (e.g. inserting to inode_map) while we are iterating over the list in dump, we'll potentially segfault.